File: 백준/Silver/1926. 그림/그림.cc
#include <iostream>
#include <queue>

using namespace std;

int n,m,cnt,maxi;
int dx[4]={0,0,1,-1};
int dy[4]={1,-1,0,0};
int arr[501][501];
int visited[501][501];
queue<pair<int,int>>q;

void bfs(){
  int check=0;
  for(int i=0;i<n;i++){
    for(int j=0;j<m;j++){
      if(arr[i][j]==1 && visited[i][j]==0){
        cnt++;
        q.push({i,j});
        visited[i][j]=1;

        while(!q.empty()){
          check++;
          int front=q.front().first;
          int back=q.front().second;
          q.pop();

          for(int k=0;k<4;k++){
            int nx=front+dx[k];
            int ny=back+dy[k];

            if(nx>=0 && ny>=0 && nx<n && ny<m && visited[nx][ny]==0 && arr[nx][ny]==1){
              q.push({nx,ny});
              visited[nx][ny]=1;
            }
          }
        }
        maxi=max(maxi,check);
        check=0;
      }
    }
  }
}
int main(){
  cin>>n>>m;
  for(int i=0;i<n;i++){
    for(int j=0;j<m;j++){
      cin>>arr[i][j];
    }
  }

  bfs();
  cout<<cnt<<endl;
  cout<<maxi;
}

File: 백준/Silver/1926. 그림/그림.java
/******************************************************************************

                            Online Java Compiler.
                Code, Compile, Run and Debug java program online.
Write your code in this editor and press "Run" button to execute it.

*******************************************************************************/
import java.util.*;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
public class Main
{
    static int[][] arr;
    static boolean[][] visited;
    static int[] dx = {1, 0, -1, 0};
    static int[] dy = {0, 1, 0, -1};
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String[] docs = br.readLine().split(" ");
		int n = Integer.parseInt(docs[0]);
		int m = Integer.parseInt(docs[1]);
		arr = new int[n][m];
		visited = new boolean[n][m];
		for(int i=0;i<n;i++){
		    String[] tmp = br.readLine().split(" ");
		    for(int j=0;j<tmp.length;j++){
		        arr[i][j] = Integer.parseInt(tmp[j]);
		    }
		}
		int cnt = 0;
		int maxi = 0;
		Queue<Pair> q = new LinkedList<>();
		for(int i=0;i<n;i++){
		    for(int j=0;j<m;j++){
		        if(arr[i][j] == 1 && !visited[i][j]){
		            int temp = 1;
		            q.add(new Pair(i, j));
		            visited[i][j] = true;
		            while(!q.isEmpty()){
		                Pair p = q.poll();
		                int x = p.x;
		                int y = p.y;
		                for(int k=0;k<4;k++){
		                    int nx = x + dx[k];
		                    int ny = y + dy[k];
		                    if(nx >= 0 && ny >= 0 && nx < n && ny < m && !visited[nx][ny] && arr[nx][ny] == 1){
		                        q.add(new Pair(nx, ny));
		                        visited[nx][ny] = true;
		                        temp++;
		                    } 
		                }
		            }
		            maxi = Math.max(maxi, temp);
		            cnt++;
		        }
		    }
		}
		System.out.println(cnt);
		System.out.println(maxi);
	}
	
	public static class Pair{
	    int x;
	    int y;
	    Pair(int x, int y){
	        this.x = x;
	        this.y = y;
	    }
	}
}